### PR TITLE
[feature] aws-okta integration

### PR DESCRIPTION
**Pull request by vivianho**
This PR:

* Adds a new `OktaProfile` option to your blessclient config
* If this OktaProfile is set, instead of using your AWS user credentials, the cmd invokes aws-okta to generate a new sts credential using Okta SAML. This assumes that you have an aws-okta profile set up (independent of the blessclient config). Most of the new code is copied from the aws-okta CLI (e.g. https://github.com/segmentio/aws-okta/blob/master/cmd/exec.go#L105-L163).

### CODE REVIEW

**Comment by vivianho** [comment]
image

**Comment by vivianho** [comment]
image

File: pkg/config/config.go [hunk 1]
```diff
@@ -65,6 +65,8 @@ type ClientConfig struct {
 	AWSUserProfile string ` yaml:"aws_user_profile"`
 	// AWSUserName is your AWS username
 	AWSUserName *string ` yaml:"aws_username,omitempty"`
+	// OktaProfile is the profile with which to invoke BLESS
+	OktaProfile *string ` yaml:"okta_profile,omitempty"`
```
Comment (edulop91) [inline pkg/config/config.go]: Might be worth making this a struct `*OktaConfig` that then holds things like the MFA,profile, and aws cred ttl

File: cmd/run.go [hunk 2]
```diff
@@ -145,6 +167,50 @@ func getAWSClient(ctx context.Context, conf *config.Config, sess *session.Sessio
...
+	}
+
+	kr, err := awsokta.OpenKeyring(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := awsokta.NewProvider(kr, profile, opts)
+	if err != nil {
+		return nil, err
```
Comment (edulop91) [inline cmd/run.go]: These can be wrapped with `errors.Wrap(err, "Some more context")` to give a bit more context to users when the errors bubble up

File: cmd/run.go [hunk 3]
```diff
@@ -145,6 +167,50 @@ func getAWSClient(ctx context.Context, conf *config.Config, sess *session.Sessio
...
+	}
+
+	creds, err := p.Retrieve()
+	if err != nil {
+		return nil, err
+	}
+
+	return &creds, nil
```
Comment (edulop91) [inline cmd/run.go]: can just `return &creds, errors.Wrap(err, "context")` and save that `if` block

File: cmd/run.go [hunk 4]
```diff
@@ -145,6 +167,50 @@ func getAWSClient(ctx context.Context, conf *config.Config, sess *session.Sessio
...
+		return nil, err
+	}
+
+	profiles, err := config.Parse()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := profiles[profile]; !ok {
+		return nil, fmt.Errorf("Profile '%s' not found in your aws config", profile)
```
Comment (edulop91) [inline cmd/run.go]: `fmt.Errorf -> errors.Errorf(...)` to get things like stack trace

**Comment by vivianho** [comment]
@edulop91 the code compiles on my laptop, but IANA go expert so I don't really understand the failure here. Please let me know how best to unblock here!

**Comment by vivianho** [comment]
@edulop91 do you want me to add unit tests on cmd/run.go? None exist for this file yet, but I can add one if you like.

**Comment by edulop91** [comment]
@vivianho thanks! that is somewhere on my TODO. As currently coded, not really possible to mock AWS and such which is why I don't really have test code there.

**Comment by edulop91** [comment]
@vivianho Since this change introduces CGO dependency (we can't really cross-compile anymore) we'll have to figure out how to release linux/darwin binaries. Seems like libusb is common enough that this won't really be a problem though. I think we could merge this and then improve from there.

**Comment by vivianho** [comment]
fwiw this is aws-okta's circleci config: https://github.com/segmentio/aws-okta/blob/master/.circleci/config.yml#L10-L13 and they have darwin/linux binaries releases

**Comment by edulop91** [comment]
https://github.com/segmentio/aws-okta#releasing gives a warning of only releasing (from a mac). Let me try releasing mac binaries from my linux machine.

**Comment by edulop91** [comment]
@vivianho lgtm. Thanks for writing all these contributions. Are you ok with me merging this?

**Pull request merged**